pkg/helm/chartproxy/repos.go

```diff
@@ -152,6 +152,13 @@ func (b helmRepoGetter) unmarshallConfig(repo unstructured.Unstructured) (*helmR
 	if err != nil {
 		return nil, err
 	}
+	tlsRefNamespace, tlsRefNamespaceFound, err := unstructured.NestedString(repo.Object, "spec", "connectionConfig", "tlsClientConfig", "namespace")
+	if err != nil {
+		return nil, err
+	}
+	if !tlsRefNamespaceFound {
+		tlsRefNamespace = configNamespace
```

Comment:
Since this is only going to trigger for Namespace scoped Repos should we default to the namespace being passed by UI. I think we should have a way to distinguish from unmarshaling  for cluster scope repo the will always go to configNamespace vs namespace scope that will either go to the one provide in CR or default to current namespace.

Reply:
The unmarshal config function is getting called for both cluster as well as namespace scoped repos.We can default to the namespace passed by ui though.